src/geometry/euclidean/circumcenter.lean

```diff
@@ -583,3 +596,75 @@ end
 end simplex
 
 end affine
+
+namespace euclidean_geometry
+
+open affine affine_subspace
+
+variables {V : Type*} {P : Type*} [inner_product_space V] [metric_space P]
+    [normed_add_torsor V P]
+include V
+
+/-- Suppose all distances from `p₁` and `p₂` to the points of a
+simplex are equal, and that `p₁` and `p₂` lie in the affine span of
+`p` with the vertices of that simplex.  Then `p₁` and `p₂` are equal
+or reflections of each other in the affine span of the vertices of the
+simplex. -/
+lemma eq_or_eq_reflection_of_dist_eq {n : ℕ} {s : simplex ℝ P n} {p p₁ p₂ : P} {r : ℝ}
+    (hp₁ : p₁ ∈ affine_span ℝ (insert p (set.range s.points)))
+    (hp₂ : p₂ ∈ affine_span ℝ (insert p (set.range s.points)))
+    (h₁ : ∀ i, dist (s.points i) p₁ = r) (h₂ : ∀ i, dist (s.points i) p₂ = r) :
+  p₁ = p₂ ∨ p₁ = reflection (affine_span ℝ (set.range s.points)) p₂ :=
+begin
```

Comment:
This proof becomes easier to read if you start with
```
set span := affine_span ℝ (set.range s.points),
```
(or some better name) and then use `span` instead of `affine_span ℝ (set.range s.points)` below.

Reply:
Or perhaps even better, add a `simplex.affine_span` `def` and use `s.affine_span` throughout

Reply:
I've made that change (with `span_s`).

Reply:
(If you have `simplex.affine_span` then I expect you'd start needing to rewrite or `change` between `simplex.affine_span` and its definition anywhere you have one, you need the other and having defeq with a reducible definition isn't enough. So I'm not sure having `simplex.affine_span` would be a good idea. There are a few other places in geometry I've avoided adding definitions for similar reasons, where they don't seem to provide enough gain to justify needing such rewrites. E.g. using `mk' p s.direction.orthogonal` instead of having a separate definition for "the subspace through `p` orthogonal to `s`", or only talking about dimensions of modules / submodules not affine spaces / subspaces.)

Reply:
The argument I'd make _for_ `simplex.affine_span` is that you might find yourself not needing to convert between `∀ i, q (s.points i)` and `∀ p ∈ set.range s.points, q p` nearly as often